cmd/build.go

```diff
@@ -193,7 +195,26 @@ func (b *buildCmd) createBuildTask() (*graph.Task, error) {
 	// TODO: create secrets
 	secrets := []*graph.Secret{}
 
-	return graph.NewTask(steps, secrets, b.opts.Registry, b.registryUser, b.registryPw, taskTotalTimeoutInSec, true)
+	var credentials []*graph.Credential
+
+	// First add the creds provided by the user
+	credentials = append(credentials, &graph.Credential{
+		RegistryName:     b.opts.Registry,
+		RegistryUsername: b.registryUser,
+		RegistryPassword: b.registryPw,
+	})
+
+	// Add any additional creds provided by the user in the --credentials flag
+	for _, creds := range b.credentials {
+		// creds should be of the form of "regName;userName;password"
+		splitCreds := strings.Split(creds, ";")
```

Comment:
Make a function for this and add unit tests. One of the tests should be `foo;bar;somepas;swordwith;semicolons` -> `[]{foo, bar, somepas;swordwith;semicols}` - will not work atm.